@SuperSandro2000, I don't know either, but it doesn't look like it has anything to do with the changes made by this PR, all 4 builds fail due to issues with dependencies (and on `darwin` pretty low level dependencies).

For `x86_64-darwin` this seems to be the issue:
```
[...]
checking run-time libs availability... failed
configure: error: one or more libs available at link-time are not available run-time. Libs used at link-time: -lnghttp2 -lssh2 -lssh2 -lssl -lcrypto -lssl -lcrypto -lgssapi_krb5 -lresolv -lz
builder for '/nix/store/j1xrg7iihy5hnhc69hjzr7x7wvd8h6sm-curl-7.82.0.drv' failed with exit code 1
cannot build derivation '/nix/store/hm5rfd4537jaij1ga0mgzhm7gl3xvynb-swift-corefoundation-unstable-2018-09-14.drv': 1 dependencies couldn't be built
[...]
```

On `aarch64-darwin`:
```
[...]
0.1.2/bin/sigtool --file "$file" check-requires-signature
Unexpected exit status from sigtool: 134
error: builder for '/nix/store/qh86d3a6a7wc5994zm1wklqcadq0fwki-python3-minimal-3.9.11.drv' failed with exit code 1;
       last 10 log lines:
       > /nix/store/q0vnphbkdcldkjrwp1xs69amf9843yjb-python3-minimal-3.9.11/lib/python3.9/config-3.9-darwin/makesetup
       > /nix/store/q0vnphbkdcldkjrwp1xs69amf9843yjb-python3-minimal-3.9.11/lib/python3.9/config-3.9-darwin/config.c
       > /nix/store/q0vnphbkdcldkjrwp1xs69amf9843yjb-python3-minimal-3.9.11/lib/python3.9/config-3.9-darwin/Setup.local
       > /nix/store/q0vnphbkdcldkjrwp1xs69amf9843yjb-python3-minimal-3.9.11/lib/python3.9/config-3.9-darwin/python.o
       > /nix/store/q0vnphbkdcldkjrwp1xs69amf9843yjb-python3-minimal-3.9.11/lib/python3.9/config-3.9-darwin/libpython3.9.a
       > dyld: Library not loaded: /nix/store/x7sy3ml65vg0h6kkkq01qmwqx5plpj8p-openssl-1.1.1n//libssl.1.1.dylib
       >   Referenced from: /nix/store/c04kqdcdfz3hrw3r2hi69lfmp6qk6mpw-sigtool-0.1.2/bin/sigtool
       >   Reason: image not found
       > /nix/store/p9v74za26sk90p313qmwmkfrm7w2sbx0-signing-utils: line 24: 37225 Abort trap: 6           /nix/store/c04kqdcdfz3hrw3r2hi69lfmp6qk6mpw-sigtool-0.1.2/bin/sigtool --file "$file" check-requires-signature
       > Unexpected exit status from sigtool: 134
       For full logs, run 'nix log /nix/store/qh86d3a6a7wc5994zm1wklqcadq0fwki-python3-minimal-3.9.11.drv'.
error: 1 dependencies of derivation '/nix/store/yx8vjnhsxq58919kr9cdl21nj6da5nam-ICU-66108.drv' failed to build
error: 1 dependencies of derivation '/nix/store/ylxrlp582pqvvi9dq72xik8pl0anjslf-libxml2-2.9.13.drv' failed to build
[...]
```

For Linux thing seem to get further. I'm going to have a look into those now.

I know that on `darwin` this change builds on `master` with #161740 cherry picked.

--- a/pkgs/development/python-modules/openai/default.nix
+++ b/pkgs/development/python-modules/openai/default.nix
@@ -9,6 +9,7 @@
 , pandas-stubs
 , requests
 , tqdm
+, wandb
 
 # Check dependencies
 , pytest-mock
@@ -35,6 +36,7 @@ buildPythonPackage rec {
     pandas-stubs
     requests
     tqdm
+    wandb
   ];
 
   pythonImportsCheck = [ "openai" ];

--- a/pkgs/development/python-modules/wandb/default.nix
+++ b/pkgs/development/python-modules/wandb/default.nix
@@ -114,6 +114,10 @@ buildPythonPackage rec {
     "tests/test_tables.py"
   ];
 
+  # Disable test that fails on darwin due to issue with python3Packages.psutil:
+  # https://github.com/giampaolo/psutil/issues/1219
+  disabledTests = lib.optional stdenv.isDarwin "test_tpu_system_stats";
+
   checkInputs = [
     azure-core
     bokeh

